### PR TITLE
cmake Improvements

### PR DESCRIPTION
**Pull request by vlabella**
Several improvements to building with cmake are included: changed .dll to .so, added install target, added static runtime option, added ability to specify apache root folder.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,25 +113,35 @@ Build instructions
...
 - Install the latest CMake from https://cmake.org/download/
 - Download Win64 Apache 2.4 from https://www.apachelounge.com/download/ (or use your own version)
-- Extract it to this folder (so there is a folder called Apache24 inside this folder)
+- note build options should match Apache build (architecture, variant, and runtime linking)
 
-**Open command prompt:**
+**cmake options:**
 
-`mkdir build`
+- USE_STATIC_RUNTIME (default is OFF) and will use /MD and /MDd. If ON will build with /MT and /MTd for release and debug, respectively.
```
Comment (TQjona) [inline README.md]: Default to ON, but recommend using it with OFF with the requirement of either
- Apache being compiled with the same Visual Studio version
- VC redist is installed

File: README.md [hunk 2]
```diff
@@ -113,25 +113,35 @@ Build instructions
...
+- APACHE_ROOT default is the local folder if not set. Otherwise set to the location of your apache instalation such as C:\Program Files\Apache24.
```
Comment (TQjona) [inline README.md]: Typo `instalation` -> `installation`

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,32 +1,56 @@
...
-if (MSVC)
-	if(CMAKE_BUILD_TYPE MATCHES Debug)
-		add_definitions(/MTd)
-	else()
-		add_definitions(/MT)
-	endif()
+if(NOT DEFINED APACHE_ROOT)
+set(APACHE_ROOT "Apache24")
+endif()
+option( USE_STATIC_RUNTIME "use MSVC static runtimes" OFF)
```
[the remark this answers is not in the document]
Comment (vlabella) [reply CMakeLists.txt]: I understand that this is a breaking change but using /MD as the default is consistent with the builds at apache lounge. In addition it is consistent with the default behavior of visual studio and cmake. VS redistributable is backward compatible now 2019 to 2017.

File: README.md (hunk 1 above)
Comment (vlabella) [reply README.md]: I understand that this is a breaking change but using /MD as the default is consistent with the builds at apache lounge. In addition it is consistent with the default behavior of visual studio and cmake. VS redistributable is backward compatible now 2019 to 2017.

File: CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (TQjona) [reply CMakeLists.txt]: Okay then it should be fine. Just fix the typos and I'll merge it. Thanks for the contribution :+1:

**Comment by vlabella** [comment]
Just pushed the typo fixes.

**Pull request merged**